clarity/src/vm/ast/parser/v2/mod.rs

```diff
@@ -894,12 +894,14 @@ impl<'a> Parser<'a> {
                             Some(expr)
                         }
                         Token::Utf8String(s) => {
-                            let mut data: Vec<Vec<u8>> = Vec::new();
-                            for ch in s.chars() {
-                                let mut bytes = vec![0; ch.len_utf8()];
-                                ch.encode_utf8(&mut bytes);
-                                data.push(bytes);
-                            }
+                            let data: Vec<Vec<u8>> = s
+                                .chars()
+                                .map(|ch| {
+                                    let mut bytes = vec![0; ch.len_utf8()];
+                                    ch.encode_utf8(&mut bytes);
```

Comment:
No problem here, but I noticed in another place with this pattern, we used `[..]` to be more explicit that it is taking a slice. Do you know which of these is considered better practice in general? The explicit slice with `[..]` seems nice to me.

Reply:
I use the `&` or `.as_slice()` when I want to explicit about using the *entire* vector, whereas I'll use the bracket notation when taking a partial slice

It's kind of like `String::from()` vs. `.to_string()` vs. `.into()`, there's no "right" answer, it depends on the context and what message exactly you want to convey.